### ISSUE TITLE
cleanco('AMBA').clean_name() is empty

### ISSUE THREAD

**Issue by dbradshaw** [psolin/cleanco#57]

**Comment by petri** [comment]
What should it be?

**Comment by dbradshaw** [comment]
I don't know. Maybe leave it as AMBA. Not familiar with the company.

**Comment by petri** [comment]
We cannot fix this if we don't know what the result should be. Would you at least know in which country this "AMBA" is in?

**Comment by dbradshaw** [comment]
No. I came across it using cleanco and it caused a problem which I had to
work around.

I don't think I can help you further. Thanks for looking into it.

-David

**Comment by FBnil** [comment]
This will always be a problem, a company that is named after a termdata (in this case from Denmark). terms_by_country might need to be limited to the countries the user expects/does business with or, the code needs to be changed that when you remove the last term, and the result is empty, rather than empty, you return the first word in the received string, as a "most probable name". To not break backwards compatibility, a new function would need to be made (or a flag might be set to change the behavior).

print(basename('AMBA')) # ""
print(basenameorfirst('AMBA')) # "AMBA"
print(basename('inc & co')) # ""
print(basenameorfirst('inc & co')) # "inc"

This would allow for companies name like 'inc & co', where even though inc is a term, it's also the name.
